webapp/src/Command/ScoreboardMergeCommand.php

```diff
@@ -0,0 +1,291 @@
+                # ignore this row.
+                if(!array_key_exists($row['team_id'], $teamIdMap)){
+                    continue;
+                }
+                $team = $teams[$teamIdMap[$row['team_id']]];
+                foreach ($row['problems'] as $problem){
+                    $label = $problem['label'];
+                    if(!array_key_exists($label, $problemIdMap)){
+                        $id = count($problems);
+                        $problemObj = new Problem();
+                        $problemObj->setProbid($id);
+                        $problemObj->setName($label);
+                        $contestProblemObj = new ContestProblem();
+                        $contestProblemObj->setProblem($problemObj);
+                        $contestProblemObj->setShortName($label);
+                        $problems[$id] = $contestProblemObj;
+                        $problemIdMap[$label] = $id;
+                        $firstSolve[$label] = Null;
+                    } else {
+                        $id = $problemIdMap[$label];
+                    }
+                    $scoreCacheObj = new scoreCache();
+                    $scoreCacheObj->setProblem($problems[$id]->getProblem());
+                    $scoreCacheObj->setTeam($team);
+                    if(array_key_exists('time', $problem)){
+                        # TODO: Make this work with input in seconds as well.
+                        $scoreCacheObj->setSolveTimePublic($problem['time']*60);
+                        $scoreCacheObj->setSolveTimeRestricted($problem['time']*60);
+                        if($firstSolve[$label] === Null or $problem['time']*60<$firstSolve[$label]){
+                            $firstSolve[$label] = $problem['time']*60;
+                        }
+                    }
+                    $scoreCacheObj->setSubmissionsPublic($problem['num_judged']);
+                    $scoreCacheObj->setSubmissionsRestricted($problem['num_judged']);
+                    $scoreCacheObj->setIsCorrectPublic($problem['solved']);
+                    $scoreCacheObj->setIsCorrectRestricted($problem['solved']);
+                    $scoreCache[] = $scoreCacheObj;
+                }
+            }
+        }
+
+        // Update the first to solve fields.
+        foreach ($scoreCache as &$scoreCacheObj){
+            if($scoreCacheObj->getSolveTimeRestricted() == $firstSolve[$scoreCacheObj->getProblem()->getName()]){
+                $scoreCacheObj->setIsFirstToSolve(true);
+            }
+        }
+
+        $scoreboard = new Scoreboard(
+            $contest,
+            $teams,
+            [$category],
+            $problems,
+            $scoreCache,
+            $freezeData,
+            /* jury = */ false,
+            $this->dj->dbconfig_get('penalty_time', 20),
+            /* scoreIsInSeconds = */ false
+        );
+
+        # Render the scoreboard to HTML and print it.
+        $data   = [
```

Comment:
I think I will try to move this (after merging) to a public function of the `ScoreBoardService` since it now reuses a lot of that code.